### DISCUSSION TITLE
osu!lazer scoring formula is different from osu! wiki

### DISCUSSION THREAD

**Discussion by 0x42697262** [ppy/osu#28203] [Q&A] [answered]
In the [scoring ruleset](https://github.com/ppy/osu/blob/9532725eb78e05b70bb89842330553a8a95514ef/osu.Game/Rulesets/Scoring/ScoreProcessor.cs#L389) for osu stable, the formula is written as:
```csharp
protected virtual double ComputeTotalScore(double comboProgress, double accuracyProgress, double bonusPortion)
        {
            return 500000 * Accuracy.Value * comboProgress +
                   500000 * Math.Pow(Accuracy.Value, 5) * accuracyProgress +
                   bonusPortion;
        }
```

However, it's different from the [wiki itself](https://github.com/ppy/osu/wiki/Scoring)

```
accuracy_portion = 0.3
combo_portion = 0.7
total_score = 1000000 * ((accuracy * accuracy_portion) + (combo / max_achievable_combo * combo_portion)) + bonus_portion
```

I believe that the wiki provides an old scoring formula? Or does is it used for Scorev2?

Also to add, how exactly does `ComputeTotalScore` works? How are `comboProgress` and `accuracyProgress` calculated?

**Comment by bdach** [comment]
~~I'm not sure where the wiki is with respect to supporting lazer facts & mechanics in general. cc @Walavouchey @ticclick~~

> Also to add, how exactly does `ComputeTotalScore` works? How are `comboProgress` and `accuracyProgress` calculated?

https://github.com/ppy/osu/blob/5e1f6964d19639b3a952ed6aa6aa1ef22dccabe1/osu.Game/Rulesets/Scoring/ScoreProcessor.cs#L371-L372

**Reply by TicClick** [reply]
https://osu.ppy.sh/wiki has very little to no details regarding the lazer scoring, and we (okay, _I_) don't maintain the lazer repo wiki, so that's on its authors

**Reply by bdach** [reply]
Oh okay I misread. Sorry, my bad, when I think of "wiki" I don't really think of this repo's wiki.

**Reply by 0x42697262** [reply]
Will the github wiki be updated for the source code's formula instead of the sv2(?) formula? If no one's gonna make a PR for it, maybe I'll make one 2-3 weeks from now.

**Reply by bdach** [reply]
> Will the github wiki be updated for the source code's formula instead of the sv2(?) formula?

I'm not sure, I'd rather just remove that entirely and redirect to source. Score is much more complicated now (due to players' demands). This sort of stuff should live in the official game wiki long-term.

> If no one's gonna make a PR for it, maybe I'll make one 2-3 weeks from now.

I'm not sure you can PR anything to a github wiki.

**Reply by 0x42697262** [reply]
Alright, thanks. I'll leave it up to you guys.

**Reply by peppy** [reply, marked as answer]
I've updated the wiki page to remove specifics and link to the source.
